build-tools/packages/build-tools/package.json

```diff
@@ -66,6 +66,7 @@
 		"sort-package-json": "1.57.0",
 		"ts-morph": "^17.0.1",
 		"type-fest": "^2.19.0",
+		"typescript": "~5.1.6",
```

Comment:
I think this is a minor pre-existing bug: because build-tools invokes the TypeScirpt compiler, I think it's a "real" dependency rather than a "devDependency".